src/components/KModal/KModal.vue

```diff
@@ -153,6 +334,27 @@ export default defineComponent({
       }
     })
 
+    const imageId = computed((): string => props.testMode ? 'test-modal-image-1234' : uuidv1())
+    const canvasId = computed((): string => props.testMode ? 'test-modal-image-canvas-1234' : uuidv1())
+    onUpdated(() => {
+      if (props.isVisible && hasHeaderImage.value && shouldComputeImageDarkness.value) {
+        // TODO: generated ID
+        const headerImageElem = document.getElementById(imageId.value) as HTMLImageElement
+        if (headerImageElem) {
+          imageStats.value.height = headerImageElem.naturalHeight
+          imageStats.value.width = headerImageElem.naturalWidth
+        }
+
+        const canvasElem = document.getElementById(canvasId.value) as HTMLCanvasElement
+        if (canvasElem) {
+          const ctx = canvasElem.getContext('2d') as CanvasRenderingContext2D
+          canvas.value = ctx
+        }
+
+        getComputedImageDarkness()
+      }
+    })
```

Comment:
Again, is all of this just to calculate the close button color..? If yes, I definitely think this is too complex for the purpose it serves.

(again, nice code but too much for just a close button)

Reply:
Removed